File: wwld_backend/demo/src/main/java/com/gateway/entity/Story.java
package com.gateway.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name="story")
@Getter
@Setter

public class Story {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Column(name="title", nullable = false)
    private String title;

    @Column(name="type")
    private Integer type;

    @Column(name="active")
    private Integer active;

    @Column(name="mainSectionId")
    private Long mainSectionId;

    @Column(name="areaId")
    private Long areaId;

    @Column(name="parentId")
    private Long parentId;

    @Column(name="time_started")
    private Date timeStarted;

    @Column(name="time_ended")
    private Date timeEnded;

    @Column(name="description")
    private String description;
}

File: wwld-next-app/src/lib/types/story.ts
export interface Story {
    id: number;
    title: string;
    description: string;
    image: string;
    type: 0 | 1; // 0: chapter, 1: screen
    mainSectionId: number;
    parentId: number;
    parentTitle: string;
}


// Interface payload
export interface StoryPayload {
  id?: number;
  title: string;
  mainSectionId: number; // ✅ sửa lại từ area_id
  description: string;
  image: string;
  type: number;
  parentId: number;
}